pkg/admission/util.go

```diff
@@ -31,38 +31,50 @@ import (
 	"github.com/apache/yunikorn-k8shim/pkg/log"
 )
 
-func updatePodLabel(pod *v1.Pod, namespace string, generateUniqueAppIds bool, defaultQueueName string) map[string]string {
-	existingLabels := pod.Labels
+func getAnnotationsForApplicationInfoUpdate(pod *v1.Pod, namespace string, generateUniqueAppIds bool, defaultQueueName string) map[string]string {
 	result := make(map[string]string)
-	for k, v := range existingLabels {
+	existingAnnotations := pod.Annotations
+	for k, v := range existingAnnotations {
 		result[k] = v
 	}
-
-	sparkAppID := utils.GetPodLabelValue(pod, constants.SparkLabelAppID)
-	appID := utils.GetPodLabelValue(pod, constants.LabelApplicationID)
-	if sparkAppID == "" && appID == "" {
+	appID := getApplicationIDFromPod(pod)
+	if appID == "" {
 		// if app id not exist, generate one
 		// for each namespace, we group unnamed pods to one single app - if GenerateUniqueAppId is not set
 		// if GenerateUniqueAppId:
 		//		application ID convention: ${NAMESPACE}-${GENERATED_UUID}
 		// else
 		// 		application ID convention: ${AUTO_GEN_PREFIX}-${NAMESPACE}-${AUTO_GEN_SUFFIX}
-		generatedID := generateAppID(namespace, generateUniqueAppIds)
-		result[constants.LabelApplicationID] = generatedID
+		appID = generateAppID(namespace, generateUniqueAppIds)
+		result[constants.AnnotationApplicationID] = appID
 
 		// if we generate an app ID, disable state-aware scheduling for this app
-		if _, ok := existingLabels[constants.LabelDisableStateAware]; !ok {
-			result[constants.LabelDisableStateAware] = "true"
+		disableStateAware := "true"
+		if value := utils.GetPodLabelValue(pod, constants.LabelDisableStateAware); value != "" {
+			disableStateAware = value
+		}
+		result[constants.AnnotationDisableStateAware] = disableStateAware
+	} else {
+		// if appID was not in pod annotations, add it to annotations.
+		if value := utils.GetPodAnnotationValue(pod, constants.AnnotationApplicationID); value == "" {
+			result[constants.AnnotationApplicationID] = appID
 		}
 	}
 
-	// if existing label exist, it takes priority over everything else
-	if _, ok := existingLabels[constants.LabelQueueName]; !ok {
+	queueName := utils.GetQueueNameFromPod(pod, "")
+	if queueName == "" {
 		// if defaultQueueName is "", skip adding default queue name to the pod labels
 		if defaultQueueName != "" {
 			// for undefined configuration, am_conf will add 'root.default' to retain existing behavior
 			// if a custom name is configured for default queue, it will be used instead of root.default
-			result[constants.LabelQueueName] = defaultQueueName
+			queueName = defaultQueueName
```

Comment:
We can just use `utils.GetQueueNameFromPod(pod, defaultQueueName)`.

Reply:
Updated it in new commit: 26dc73107f5206ad7cdd021e1af8635d9efe803d